### ISSUE TITLE
Create asset library schema in Sanity

### ISSUE THREAD

**Issue by SvSven** [equinor/energyvision#670]
See: https://github.com/equinor/energyvision/issues/536

Set up the schema in Sanity for the custom file management system.
Files we initially want to support for this approach is PDFs and Excel sheets.

Potentially we may want to move SVG illustrations and icons to this system as well

**Comment by SvSven** [comment]
I've now added a first version of a file asset library: 

![image.png](https://images.zenhubusercontent.com/5dba9d22e2edb600015fe6c0/d785926a-1ce2-482b-8c62-166a5ac6f94d)

Currently this is limited to the following file types: .pdf, .xls, .xlsx, .csv
but we may want to expand that later on by adding for example illustrations and icons

The form is currently pretty straightforward as well:

![image.png](https://images.zenhubusercontent.com/5dba9d22e2edb600015fe6c0/0e920c41-12d9-40ea-89b6-c7b7043b7b70)

Only the title is currently a mandatory field, the file upload/select is not mandatory. This is to allow the use case where a file could contain time-sensitive information; the editor can then make this asset document in advance (with title, tags, etc) and link it to the relevant pages, and then upload the file attachment once it is ready to be made public.

It should be noted that once you upload a file to the Sanity DAM it becomes available to all editors, and can be queried from the web if you know how/where to do so. We're checking with Sanity if there is any way of restricting access to uploaded files or not. Until we have an answer or solution for that it comes down to: don't upload files to Sanity that are not supposed to be public

The tags for the assets can be made/edited in the "Manage file tags" menu item under the asset library, as well as when adding them to a file document directly (which is a new feature in Sanity)

![image.png](https://images.zenhubusercontent.com/5dba9d22e2edb600015fe6c0/c4de761e-a6e7-43be-8aca-4bcceb1f8d4c)

Adding tags will allow editors to filter the list of files by tag. In addition to that I created a predefined filter to show all excel sheets (xls, xlsx, csv) and all pdf files

![image.png](https://images.zenhubusercontent.com/5dba9d22e2edb600015fe6c0/c9a750f8-2939-459c-b91c-1433185ae55b)

These tag filters are automatically added once you create a new tag.

**Comment by nilsml** [comment]
What about zip-files? Assess security (we have a separate issue for that)

**Comment by SvSven** [comment]
@meols @nicholas-haggerty 

Some answers to feedback and questions:

**Additional file types**
We can add support for more file types if needed, that should not be an issue. However things like ZIP files could be a potential security risk.

**Assets, tags & satellite sites**
Both assets and asset tags are individual to each site; meaning what you make in the global site is not shared with any of the satellite sites.

**Filtering with multiple tags**
Currently it's only possible to select one tag at a time in the document list. I've checked with Sanity and they have a feature request to be able to do more filtering, sorting, ordering in these document lists but there is no timeline for that yet. Don't have a good solution for this yet, but will look a bit further into it to see what we can do.

**Replacing a file**
When you replace a file, the previous version will still be available in the media library until manually deleted; there is currently nothing set up to delete the previously used file. We may be able to set something up to handle that automatically but I'm not sure if that's possible or not.

**Original filename**
We can add the original filename to the end of the URL like Sanity suggested, which could then look something like this:
  - https://cdn.sanity.io/images/h61q9gi9/production/816ff0d2931fd53aca63b6bba91f50c3209da9d3-491x289.svg&filename=CEO_reflections_Infographics-2-ENG
  - https://cdn.sanity.io/images/h61q9gi9/production/816ff0d2931fd53aca63b6bba91f50c3209da9d3-491x289.svg?CEO_reflections_Infographics-2-ENG

But bare in mind that you then need to be fully aware of what the filename is before you upload the file, as you cannot change the original filename once it's been uploaded. The question is also what kind of problem it solves.

**Google Analytics tracking**
We currently don't have anything set up for GA yet so that's a good question. Out of the box it would track the link clicked, so it would register the sanity CDN link. Once we get started with integrating GA however we could set up a custom event for when these kind of links are clicked with more details related to the file.

**Comment by SvSven** [comment]
@meols @nicholas-haggerty 

Further feedback from Sanity:

**Access to files**
> As soon as an asset is uploaded it is public. Whether it’s referenced by a document or not. Anyone could theoretically guess the filename of the asset (which is a hash based on the original image’s dimensions, filename, etc) but the likelihood of that is extremely low.

**CDN**
> It would be possible for Sanity to explicitly request the purging of a specific asset from the CDN, which we would need to perform on an as-requested basis. Also it’s possible that a file can live on our CDN for a year, as that’s the TTL of the file.

Let me know if there is anything else.

**Comment by nilsml** [comment]
Can we close this one now @meols @nicholas-haggerty? We cannot leave this issue in review for forever.

**Comment by nicholas-haggerty** [comment]
@nilsml No. I suggest you move to another lane. Still issues that need clarifying.

**Comment by nicholas-haggerty** [comment]
@SvSven @meols Lets grab 20 mins next week to hopefully finalize the asset concerns

**Comment by SvSven** [comment]
- Add filename to end of Sanity asset URL
 - Look into adding hover tooltop to download list (EDS?)
- Figure out a way of hiding documents in media plugin
  - Can we limit uploading in the media plugin to images only? Exclude files that are meant for the asset library system
- Is it a risk to upload files to Sanity that should not be available to the public yet?
  - Good place to test new feature to link to draft content? Check with Sanity for feedback

File types to add:
- ZIP
- ACS

Won't be used often but for the cases they are needed it should be supported. Don't need separate filter for it, can just use a "Other filetypes" filter or something.

**Comment by nilsml** [comment]
Link to meeting notes: https://github.com/equinor/energyvision-internal/discussions/14

**Comment by nilsml** [comment]
Hiding files covered by #723

**Comment by SvSven** [comment]
Adding additional file types: #806
